src/API/AbstractAPIClient.php

```diff
@@ -41,25 +41,17 @@ public function callAPI(Request $request)
 
             $method = $request->getMethod();
             $params = $request->getParameters();
-            $url = $request->getUrl();
-            $headers = $request->getHeaders();
-            $body = $request->getBody();
-
-            $isPaginatable = false;
-            if ($method === 'GET') {
-                $isPaginatable = true;
-            }
 
             $mergedResponse = null;
 
             $currentPage = 1;
             $totalPages = 1;
 
             while ($totalPages >= $currentPage) {
```

Comment:
If we call pagination **after** the request the code will have duplicates. From my perspective, it doesn't matter so if you think it's more readable with duplicate code I can do that.